File: src/python/maps/map09.py
def to8(engineRef):
    yield from engineRef.mapSwitchTask('map08.ika-map', (1 * 16, engineRef.player.sprite.y))

def to10(engineRef):
    yield from engineRef.mapSwitchTask('map10.ika-map', (engineRef.player.sprite.x, 28 * 16))

File: src/winter/maps/map09.ts
import { MapScript } from "./mapscript.js"
import { PyEngine } from "./../winter.js"

export default new MapScript(autoexec, { to8, to10 })

export function autoexec(_engineRef: PyEngine) {}

export function *to8(engineRef: PyEngine) {
    const p = engineRef.getPlayerEntity()
    yield* engineRef.mapSwitchTask('map08.ika-map', [1 * 16, p.sprite.y])
}

export function *to10(engineRef: PyEngine) {
    const p = engineRef.getPlayerEntity()
    yield* engineRef.mapSwitchTask('map10.ika-map', [p.sprite.x, 28 * 16])
}
